### PR TITLE
Fix misleading log statements. Fix potential NPE.

### PR DESCRIPTION
**Pull request by iocanel**

### CODE REVIEW

**Comment by iocanel** [comment]
@jimmidyson: modified test case, so that it also goes throguh waitForScale().

**Comment by iocanel** [comment]
Hmmm, I don't see the added commits though.

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/DeploymentOperationsImpl.java [hunk 1]
```diff
@@ -137,9 +140,11 @@ public void run() {
...
     try {
       if (Utils.waitUntilReady(queue, getConfig().getScaleTimeout(), TimeUnit.MILLISECONDS)) {
-        int currentReplicas = atomicDeployment.get().getStatus().getReplicas() != null ? atomicDeployment.get().getStatus().getReplicas() : 0;
-        LOG.error("Only {}/{} pod(s) ready for Deployment: {} in namespace: {} - giving up",
-          currentReplicas, atomicDeployment.get().getSpec().getReplicas(), atomicDeployment.get().getMetadata().getName(), namespace);
+        LOG.debug("{}/{} pod(s) ready for Deployment: {} in namespace: {}.",
+          replicasRef.get(), count, name, namespace);
+      } else {
+        LOG.error("{}/{} pod(s) ready for Deployment: {} in namespace: {}  after waiting for {} seconds so giving up",
+          replicasRef.get(), count, name, namespace, getConfig().getScaleTimeout(), TimeUnit.MILLISECONDS);
```
Comment (jimmidyson) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/DeploymentOperationsImpl.java]: Wrong logging here for TimeUnit.MILLISECONDS - assume you're trying to convert the scale timeout in millis to seconds? `TimeUnit.MILLISECONDS.toSeconds(getConfig().getScaleTimeout())` perhaps?

File: openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/DeploymentConfigOperationsImpl.java [hunk 2]
```diff
@@ -207,9 +210,11 @@ public void run() {
...
       try {
         if (Utils.waitUntilReady(queue, getConfig().getScaleTimeout(), TimeUnit.MILLISECONDS)) {
-          int currentReplicas = atomicDeploymentConfig.get().getStatus().getReplicas() != null ? atomicDeploymentConfig.get().getStatus().getReplicas() : 0;
-          LOG.error("Only {}/{} pod(s) ready for DeploymentConfig: {} in namespace: {} - giving up",
-            currentReplicas, atomicDeploymentConfig.get().getSpec().getReplicas(), atomicDeploymentConfig.get().getMetadata().getName(), namespace);
+          LOG.debug("{}/{} pod(s) ready for DeploymentConfig: {} in namespace: {}.",
+            replicasRef.get(), count, name, namespace);
+        } else {
+          LOG.error("{}/{} pod(s) ready for DeploymentConfig: {} in namespace: {}  after waiting for {} seconds so giving up",
+            replicasRef.get(), count, name, namespace, getConfig().getScaleTimeout(), TimeUnit.MILLISECONDS);
```
Comment (jimmidyson) [inline openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/DeploymentConfigOperationsImpl.java]: Same again.

**Comment by iocanel** [comment]
[merge]

**Pull request merged**